### PR TITLE
update scd docs per feedback

### LINKED ISSUE #5395: Update SL Dimensions SCD section to comply with the validation rules
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/docs/build/dimensions

### What part(s) of the page would you like to see updated?

Current'y it's not very clear how to properly use SCD dimensions:  
1. In the top table in `type` field we say "There are three types", but in YML there are only 2 options possible, which is fine, but quite confusing
2. In all dimension examples it's shown `metric_time` as a dimension, however it's a reserved word and such dimension is not allowed
3. `Note: The SCD dimensions table must have valid_to and valid_from columns.` -- it's not clear if these columns are really physically should exist or just configured via `validity_params : is_start / is_end`

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by mirnawong1**
Opened this pr to address the feedback from issue #5395.  It also adds info regarding validity_params and how it relates to the scd table columns: valid_to/valid_from. 

[ ] Outstanding question wrt  metric_time
[ ] PM review needd

Resolves #5395

### CODE REVIEW

File: website/docs/docs/build/dimensions.md [hunk 1]
```diff
@@ -45,8 +45,8 @@ semantic_models:
     description: A record for every transaction that takes place. Carts are considered multiple transactions for each SKU. 
     model: {{ ref("fact_transactions") }}
     defaults:
-      agg_time_dimension: metric_time
-# --- entities ---
+      agg_time_dimension: metric_time
```
Comment (mirnawong1) [inline website/docs/docs/build/dimensions.md]: should this be `metric_time`?

Comment (Jstein77) [reply website/docs/docs/build/dimensions.md]: No it should be 'order_date'

**Review by Jstein77 — APPROVED** [review]
This is more clear. I would love to add a cook book example that has a more real life scenario. I think this is a good explanation, but i still think it will be hard for people to adopt SCD dimensions just from reading this.

File: website/docs/docs/build/dimensions.md (hunk 1 above)
Comment (mirnawong1) [inline website/docs/docs/build/dimensions.md]:
```suggestion
      agg_time_dimension: order_date
```

**Pull request merged**